rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -1960,6 +1960,46 @@ def test_upgrade_db_39_to_40(user_data_dir):  # pylint: disable=unused-argument
     ).fetchall() == [(bnb_velo_asset_id,)]
 
 
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_upgrade_db_40_to_41(user_data_dir):
+    """Test upgrading the DB from version 40 to version 41"""
+    msg_aggregator = MessagesAggregator()
+    _use_prepared_db(user_data_dir, 'v40_rotkehlchen.db')
+    db_v40 = _init_db_with_target_version(
+        target_version=40,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+        resume_from_backup=False,
+    )
+    cursor = db_v40.conn.cursor()
+    assert table_exists(cursor, 'key_value_cache') is False
+    cursor.close()
+    db_v40.logout()
+
+    # Execute upgrade
+    db = _init_db_with_target_version(
+        target_version=41,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+        resume_from_backup=False,
+    )
+    cursor = db.conn.cursor()
+    assert table_exists(cursor, 'key_value_cache') is True
+    cursor.execute('SELECT sql from sqlite_master WHERE type="table" AND name="key_value_cache"')
```

Comment:
This is for checking the schema? If so can you maybe add another optional argument to `table_exists` that is called `schema` and if passed check the schema too?